integration/handle-error-with-after-response/simple.ts

```diff
@@ -0,0 +1,188 @@
+}
+
+export const GetBasicResponse = {
+  encode(message: GetBasicResponse, writer: _m0.Writer = _m0.Writer.create()): _m0.Writer {
+    if (message.name !== "") {
+      writer.uint32(10).string(message.name);
+    }
+    return writer;
+  },
+
+  decode(input: _m0.Reader | Uint8Array, length?: number): GetBasicResponse {
+    const reader = input instanceof _m0.Reader ? input : _m0.Reader.create(input);
+    let end = length === undefined ? reader.len : reader.pos + length;
+    const message = createBaseGetBasicResponse();
+    while (reader.pos < end) {
+      const tag = reader.uint32();
+      switch (tag >>> 3) {
+        case 1:
+          if (tag !== 10) {
+            break;
+          }
+
+          message.name = reader.string();
+          continue;
+      }
+      if ((tag & 7) === 4 || tag === 0) {
+        break;
+      }
+      reader.skipType(tag & 7);
+    }
+    return message;
+  },
+
+  fromJSON(object: any): GetBasicResponse {
+    return { name: isSet(object.name) ? globalThis.String(object.name) : "" };
+  },
+
+  toJSON(message: GetBasicResponse): unknown {
+    const obj: any = {};
+    if (message.name !== "") {
+      obj.name = message.name;
+    }
+    return obj;
+  },
+
+  create<I extends Exact<DeepPartial<GetBasicResponse>, I>>(base?: I): GetBasicResponse {
+    return GetBasicResponse.fromPartial(base ?? ({} as any));
+  },
+  fromPartial<I extends Exact<DeepPartial<GetBasicResponse>, I>>(object: I): GetBasicResponse {
+    const message = createBaseGetBasicResponse();
+    message.name = object.name ?? "";
+    return message;
+  },
+};
+
+export interface BasicService {
+  GetBasic(request: GetBasicRequest): Promise<GetBasicResponse>;
+}
+
+export const BasicServiceServiceName = "basic.BasicService";
+export class BasicServiceClientImpl implements BasicService {
+  private readonly rpc: Rpc;
+  private readonly service: string;
+  constructor(rpc: Rpc, opts?: { service?: string }) {
+    this.service = opts?.service || BasicServiceServiceName;
+    this.rpc = rpc;
+    this.GetBasic = this.GetBasic.bind(this);
+  }
+  GetBasic(request: GetBasicRequest): Promise<GetBasicResponse> {
+    try {
+      const data = GetBasicRequest.encode(request).finish();
+      const promise = this.rpc.request(this.service, "GetBasic", data);
+      return promise.then((data) => {
+        try {
+          const response = GetBasicResponse.decode(_m0.Reader.create(data));
+          if (this.rpc.afterResponse) {
+            this.rpc.afterResponse(this.service, "GetBasic", response);
+          }
+          return response;
+        } catch (error) {
+          if (error instanceof Error && this.rpc.handleError) {
+            throw this.rpc.handleError(this.service, "GetBasic", error);
+          }
+          throw error;
+        }
+      });
+    } catch (error) {
+      if (error instanceof Error && this.rpc.handleError) {
+        throw this.rpc.handleError(this.service, "GetBasic", error);
+      }
+      throw error;
```

Comment:
@lukealvoeiro wdyt of instead of `throw error`, we make this `return Promise.reject(error)`?

My rationale is that, as-is, because this `GetBasic` method isn't using the `async` keyword, errors that are thrown won't actually turn into rejected promises, which is typically what callers except.

Reply:
ah true, good suggestion - will make this change!